index.html

```diff
@@ -197,9 +211,44 @@
       });
     });
 
+    //specifications for the plugins in the clinicaTimeline
+    var clinicalTimelineSpecifications = {
+      configButton : {
+        trimTimeline: {text: "Trimming", bool: "enableTrimmedTimeline"},
+        overviewAxis: {text: "Overiew Axis", bool: "enableOverviewAxis"},
+        helperLines: {text: "Helper Lines", bool: "enableHelperLines"},
+        verticalLine: {text: "Vertical Line", bool: "enableVerticalLine"},
+        zoom: {text: "Zooming", bool: "enableZoom"},
+        exportTimeline: {text: "Export", bool: null},
+        views: {text: "Tabbed Views", bool: null}
+      }
+    };
+
+    //plugins for clinicalTimeline
+    var clinicalTimelinePlugins = {
+      trimTimeline: {
+        bool: "enableTrimmedTimeline", func: trimClinicalTimeline, spec: {}
+      },
+      verticalLine: {
+        bool: "enableVerticalLine", func: clinicalTimelineVerticalLine, spec: {}
+      },
+      helperLines: {
+        bool: "enableHelperLines", func: initHelperLines, spec: {}
+      },
+      overviewAxis: {
+        bool: "enableOverviewAxis", func: clinicalTimelineOverviewAxis, spec: {}
+      },
+      configButton: {
+        bool: null, func: configCheckManager, spec:clinicalTimelineSpecifications.configButton 
+      },
+      views: {
+        bool: null, func: clinicalTimelineViewHandler, spec:clinicalTimelineSpecifications.configButton
+      }
+    };
```

Comment:
I think this whole block can be an array of plugins, instead of passing specific functions. All the plugins should have the same function signature, so `func:` is not necessary. It would always be the same. I don't think the `bool` is necessary either, since you are passing the plugins as arguments to `timeline.plugins()`. Thereby the user can already enable/disable plugins programmatically. Within the configButton plugin you could remove/add plugins from this plugins array. 

The `spec:` is necessary to configure the plugin, but I think that's better passed as argument to the plugin directly instead of through `clinicalTimeline.js`. The plugins constructor could take some configuration object, then the returned object can be given to `timeline.plugins()`.